tt.go

```diff
@@ -0,0 +1,282 @@
+}
+
+func (*StringToTTConvFactory) MakeTypeToDatetimeConverter() Converter[string, any] {
+	return MakeStringToDateTimeConverter()
+}
+
+func (*StringToTTConvFactory) MakeTypeToUUIDConverter() Converter[string, any] {
+	return MakeStringToUUIDConverter()
+}
+
+func (*StringToTTConvFactory) MakeTypeToMapConverter() Converter[string, any] {
+	return MakeStringToMapConverter()
+}
+
+func (*StringToTTConvFactory) MakeTypeToArrayConverter() Converter[string, any] {
+	return MakeStringToSliceConverter()
+}
+
+func (*StringToTTConvFactory) MakeTypeToVarbinaryConverter() Converter[string, any] {
+	return MakeStringToBinaryConverter()
+}
+
+func (fac *StringToTTConvFactory) MakeTypeToDoubleConverter() Converter[string, any] {
+	return MakeStringToFloatConverter(fac.thousandSeparators, fac.decimalSeparators)
+}
+
+func (fac *StringToTTConvFactory) MakeTypeToDecimalConverter() Converter[string, any] {
+	return MakeStringToDecimalConverter(fac.thousandSeparators, fac.decimalSeparators)
+}
+
+func (fac *StringToTTConvFactory) MakeTypeToIntegerConverter() Converter[string, any] {
+	return MakeSequenceConverter([]Converter[string, any]{
+		MakeStringToUIntConverter(fac.thousandSeparators),
+		MakeStringToIntConverter(fac.thousandSeparators),
+	})
+}
+
+func (fac *StringToTTConvFactory) MakeTypeToNumberConverter() Converter[string, any] {
+	return MakeSequenceConverter([]Converter[string, any]{
+		MakeStringToUIntConverter(fac.thousandSeparators),
+		MakeStringToIntConverter(fac.thousandSeparators),
+		MakeStringToFloatConverter(fac.thousandSeparators, fac.decimalSeparators),
+	})
+}
+
+func (fac *StringToTTConvFactory) MakeTypeToIntervalConverter() Converter[string, any] {
+	return MakeStringToIntervalConverter()
+}
+
+func (fac *StringToTTConvFactory) MakeTypeToAnyConverter() Converter[string, any] {
+	return MakeSequenceConverter([]Converter[string, any]{
+		fac.MakeTypeToNumberConverter(),
+		fac.MakeTypeToDecimalConverter(),
+		fac.MakeTypeToBooleanConverter(),
+		fac.MakeTypeToDatetimeConverter(),
+		fac.MakeTypeToUUIDConverter(),
+		fac.MakeTypeToIntervalConverter(),
+		fac.MakeTypeToStringConverter(),
+	})
+}
+
+func (fac *StringToTTConvFactory) MakeTypeToScalarConverter() Converter[string, any] {
+	return MakeSequenceConverter([]Converter[string, any]{
+		fac.MakeTypeToNumberConverter(),
+		fac.MakeTypeToDecimalConverter(),
+		fac.MakeTypeToBooleanConverter(),
+		fac.MakeTypeToDatetimeConverter(),
+		fac.MakeTypeToUUIDConverter(),
+		fac.MakeTypeToIntervalConverter(),
+		fac.MakeTypeToStringConverter(),
+	})
+}
+
+func (fac *StringToTTConvFactory) MakeNullableConverter(
+	converter Converter[string, any]) Converter[string, any] {
+	return MakeSequenceConverter([]Converter[string, any]{
+		MakeStringToNullConverter(fac.nullValue),
+		converter,
+	})
+}
+
+// WithNullValue sets nullValue.
+func (fac *StringToTTConvFactory) WithNullValue(nullValue string) *StringToTTConvFactory {
+	fac.nullValue = nullValue
+	return fac
+}
```

Comment:
```suggestion
// WithNullValue sets nullValue.
func (fac StringToTTConvFactory) WithNullValue(nullValue string) *StringToTTConvFactory {
	fac.nullValue = nullValue
	return fac
}
```

The type could be immutable with the patter:
```Go
factory := MakeFactory().WithOpt1().WithOpt2()
```

Reply:
Changed.